src/components/organisms/PageHeader/K8sVersionSelection.tsx

```diff
@@ -0,0 +1,67 @@
+import {Dropdown, Menu, Typography} from 'antd';
+
+import {useAppDispatch, useAppSelector} from '@redux/hooks';
+import {updateK8sVersion} from '@redux/reducers/appConfig';
+import {downloadK8sSchema} from '@redux/thunks/downloadK8sSchema';
+
+import {doesSchemaExist} from '@utils/index';
+
+import {K8S_VERSIONS} from '@shared/constants';
+
+import * as S from './K8sVersionSelection.style';
+
+export const K8sVersionSelection = () => {
+  const dispatch = useAppDispatch();
+  const appConfig = useAppSelector(state => state.config);
+  const userDataDir = useAppSelector(state => state.config.userDataDir);
+
+  const selectedK8SVersion = String(appConfig?.k8sVersion);
```

Comment:
I recommend the following change here because if we select the entire `state.config` object, then this component will re-render each time any of it's properties change. However, if we select the `k8sVersion` directly, then the component will re-render only when that changes.

```suggestion
  const selectedK8SVersion = useAppSelector(state => String(state.config.k8sVersion));
  const userDataDir = useAppSelector(state => state.config.userDataDir);
```